cmake/Dependencies.cmake

```diff
@@ -1679,6 +1679,10 @@ if(NOT INTERN_BUILD_MOBILE)
     # we want to respect the standard, and we are bored of those **** .
     add_definitions(-D_CRT_SECURE_NO_DEPRECATE=1)
     string(APPEND CMAKE_CUDA_FLAGS " -Xcompiler=/wd4819,/wd4503,/wd4190,/wd4244,/wd4251,/wd4275,/wd4522")
+    # Want to be able to capture constexpr in lambdas and this flag.
+    # The /Zc:lambda compiler option enables the conforming lambda processor in MSVC.
+    # Note that this will be turned on by defualt for std=c++20 and above
+    string(APPEND CMAKE_CUDA_FLAGS " -Xcompiler /Zc:lambda")
```

Comment:
@malfet adding this compilation flag seems to have enabled the building of the attention/ files for msvc. However it seems that the updated lambda processor breaks others:

Current build failure:
``` Bash
2023-01-15T04:32:03.7577420Z C:\actions-runner\_work\pytorch\pytorch\aten\src\ATen\native\quantized\cuda\AffineQuantizer.cu(45): error C2326: 'scalar_t at::native::_GLOBAL__N__809cb215_18_AffineQuantizer_cu_95ed4547_5828::quantize_tensor_per_tensor_affine_cuda::<lambda_1>::()::<lambda_1>::()::<lambda_1>::operator ()(float,scalar_t) const': function cannot access 'qmin'
```

I am curious if what you think would be the best step forward. Trying to chase the rabbit fixing each compilation error with the flag global. Or trying to selectively apply this flag to just the files that need it (I am not sure the best way to do this).

Zc:lambda will be turned on by default when we make the migration to c++20
https://learn.microsoft.com/en-us/cpp/build/reference/zc-lambda?view=msvc-170

Reply:
You can apply the flag to a specific file, by using `set_source_file_properties`, something akin to
https://github.com/pytorch/pytorch/blob/ef2586422cbe3e8e2ee56478588a5bbc699d49a3/caffe2/CMakeLists.txt#L538-L539